### ISSUE TITLE
Saving msoffice files failes

### ISSUE THREAD

**Issue by robert-boulanger** [Scille/winfspy#25]
When saving a ms office 365 file (Word, Excel) the provided memfs crashes immediatly.

So far I could target the error in  in 
winstuff.py 
def evolve

The return value of **lib.FspSetSecurityDescriptor** is not checked there.
Changing the method to:

```
def evolve(self, security_information, modification_descriptor):
    psd = ffi.new("SECURITY_DESCRIPTOR**")
    if NTSTATUS.STATUS_SUCCESS == lib.FspSetSecurityDescriptor(
        self.handle, security_information, modification_descriptor, psd):
        handle = psd[0]
        size = lib.GetSecurityDescriptorLength(handle)
        return type(self)(handle, size)
    else:
        raise RuntimeError(
            f"Cannot create new security descriptor"
            f"{cook_ntstatus(lib.GetLastError())}"
        )
```

solves the problem, but I'm not quiet sure if this is enough so far.

**Comment by vxgmichel** [comment]
Hi @robert-boulanger and thanks a lot for the report,

I'll investigate this issue and let you know what I find.

**Comment by vxgmichel** [comment]
Sadly I haven't been able to reproduce your issue on my windows machine. However, the fix you suggest makes sense and I'll create a dedicated PR.

Do you have a log you could share with the `RuntimeError` you added? I'd like to see the stack strace for this issue.

**Comment by robert-boulanger** [comment]
This issue appeared on a Win10 Enterprise. Maybe Word behaves different on Prof or Home Editions.
It appeared when editing the word file and closing the application (without saving before) 
Then a dialog pops up asking whether to save or not. There I pressed save.

[winfspy_word_save_crash.txt](https://github.com/Scille/winfspy/files/5034980/winfspy_word_save_crash.txt)

**Comment by robert-boulanger** [comment]
And here the log with the provided fix
(BTW: A Added a BEG line to the wrapper which gets printed before the function is executed)
[winfspy_word_save_fixed.txt](https://github.com/Scille/winfspy/files/5034999/winfspy_word_save_fixed.txt)

**Comment by vxgmichel** [comment]
Thanks for the logs, I have been able to reproduce the issue.

I'm working on a fix at the moment.

**Comment by vxgmichel** [comment]
Fixed in PR #26, released in [version 0.8.0](https://github.com/Scille/winfspy/releases/tag/v0.8.0)

Thanks again!
